overlord/devicestate/systems.go

```diff
@@ -83,28 +83,39 @@ func systemFromSeed(label string) (*System, error) {
 	if err != nil {
 		return nil, fmt.Errorf("cannot obtain brand: %v", err)
 	}
-	system := System{
+	system := &System{
 		Current: false,
 		Label:   label,
 		Model:   model,
 		Brand:   brand,
 		Actions: defaultSystemActions,
 	}
-	return &system, nil
+	if currentSystem != nil && isCurrentSystem(currentSystem, system) {
+		system.Current = true
+		system.Actions = currentActions
+	}
+	return system, nil
 }
 
-func currentSystemForMode(st *state.State, mode string) (*seededSystem, error) {
+func currentSystemForMode(st *state.State, mode string) (currentSys *seededSystem, actions []SystemAction, err error) {
 	switch mode {
 	case "run":
-		return currentSeedSystem(st)
+		actions = currentSystemActions
+		currentSys, err = currentSeedSystem(st)
 	case "install":
 		// there is no current system for install mode
-		return nil, nil
+		return nil, nil, nil
 	case "recover":
+		actions = recoverSystemActions
 		// recover mode uses modeenv for reference
-		return seededSystemFromModeenv()
+		currentSys, err = seededSystemFromModeenv()
+	default:
+		return nil, nil, fmt.Errorf("internal error: cannot identify current system for unsupported mode %q", mode)
+	}
+	if err != nil {
+		return nil, nil, err
 	}
-	return nil, fmt.Errorf("internal error: cannot identify current system for unsupported mode %q", mode)
+	return currentSys, actions, nil
 }
 
 func currentSeedSystem(st *state.State) (*seededSystem, error) {
```

Comment:
maybe for consistency this should be called: currentSeededSystem ?

Reply:
this isn't renamed yet

Reply:
Missed that, should be good now.